### ISSUE TITLE
Expose CLI parsing logic that converts "cypress run" arguments to options object

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#7760]
It would be nice to expose a small utility from Cypress NPM module to parse "cypress run" CLI arguments into options object. Could be used to wrap Cypress and get for example JSON of results (#7532) while running everything the same way as "normal run" does

We already have the code in `cli/lib/cli.js`, just need to expose it from the NPM module

```js
// user code
const cypress = require('cypress')
const runOptions = cypress.cli.parseRunArguments(process.argv.slice(2))
cypress.run(runOptions).then(...)
```

Then it could eliminate need to prefix `cypress run` commands like Percy or Happo plugins do right now

```
npx percy-cypress -- cypress run ...
```

and would allow those plugins to receive the Dashboard URL to link their run to Cypress Dashboard run, like done in https://github.com/cypress-io/github-action#outputs

┆Issue is synchronized with this [Jira Story](https://cypress-io.atlassian.net/browse/TR-103) by [Unito](https://www.unito.io/learn-more)

**Comment by bahmutov** [comment]
It could also enable checking Cypress test run against expected number of tests, their results, etc, Almost like meta testing

**Comment by jennifer-shehane** [comment]
Released in [`4.11.0`](https://on.cypress.io/changelog#4-11-0).

This comment thread has been locked. If you are still experiencing this issue after upgrading to
Cypress v4.11.0, please open a new issue.
